packages/core/useOnceInAllTabs/index.ts

```diff
@@ -0,0 +1,89 @@
+import type { AnyFn } from '@vueuse/shared'
+import type { ComputedRef, Ref } from 'vue'
+import type { ConfigurableNavigator } from '../_configurable'
+import { tryOnMounted, tryOnScopeDispose } from '@vueuse/shared'
+import { readonly, ref } from 'vue'
+import { defaultNavigator } from '../_configurable'
+import { useSupported } from '../useSupported'
+
+export interface UseOnceInAllTabsOptions<Keys extends string = string> extends ConfigurableNavigator {
+  /**
+   * The name of the lock that gets used to elect the one tab that does the work.
+   * Needs to be uniq throughout the origin.
+   */
+  name: Keys
+}
+
+export interface UseOnceInAllTabsReturn<Workload extends AnyFn = () => unknown> {
+  /**
+   * Returns whether the Web Locks API is supported by the current browser.
+   */
+  isSupported: ComputedRef<boolean>
+
+  /**
+   * Reactive boolean that tracks whether the current tab was elected as the one tab to do the work.
+   * If the tab was elected as leader it will stay the leader until the current scope gets disposed.
+   * Then a new tabs becomes the leader. isElected will only ever transition from `false` to `true` in
+   * the lifetime of a component.
+   */
+  isElected: Readonly<Ref<boolean>>
+
+  /**
+   * Executes the workload function if the Web Locks API is not supported or if the current tab is the elected tab.
+   *
+   * @return `true` when the workload function was executed `false` otherwise.
+   */
+  doOnce: (workload: Workload) => boolean
+}
+
+/**
+ * Ensure that code gets executed only in one of many tabs.
+ *
+ * @see https://vueuse.org/core/useOnceInAllTabs/
+ * @param [options]
+ */
+export function useOnceInAllTabs<Workload extends AnyFn = AnyFn, Keys extends string = string>(options: UseOnceInAllTabsOptions<Keys>): UseOnceInAllTabsReturn<Workload> {
+  const {
+    name,
+    navigator = defaultNavigator,
+  } = options
+  const isSupported = useSupported(() => navigator?.locks)
```

Comment:
https://github.com/vueuse/vueuse/pull/4514

We should include the existence of AbortController to supported.

Reply:
OK, did that. Anything that currently supports the [Web Locks API](https://caniuse.com/mdn-api_navigator_locks) does support the [AbortController](https://caniuse.com/abortcontroller), too.

Reply:
> OK, did that. Anything that currently supports the [Web Locks API](https://caniuse.com/mdn-api_navigator_locks) does support the [AbortController](https://caniuse.com/abortcontroller), too.

This is true. But I think this will help with future regressions if for some reason the AbortController is not available (bug, security settings, new runtime/browser ...).